dvc/cli.py

```diff
@@ -108,6 +142,13 @@ def error(self, message, cmd_cls=None):  # pylint: disable=arguments-differ
         _find_parser(self, cmd_cls)
 
     def parse_args(self, args=None, namespace=None):
+        if args is None:
+            args = sys.argv[1:]
+        else:
+            args = list(args)
+
+        self._find_cmd_suggestions(args)
```

Comment:
Can we do this when we run into an error instead of doing this in advance?

Reply:
Looks like currently we pretty much do the arg validation instead of argparse and raise an error if we don't recognise the command. Could we only kick this when argparse reports the error?

Reply:
The code now tries to find command suggestions when `argparse` throws an error. This currently only works for single commands (e.g. `dvc ad`) and not subcommands (e.g. `dvc remote modiyf`) so I still need to figure that out.

Raising an error when we don't recognize a command requires setting `exit_on_error=False` when initializing DvcParser. This requires Python 3.9.

More info can be found here: https://docs.python.org/3/library/argparse.html#exit-on-error

Reply:
Okay, I was looking at the wrong `add_parser()` method for subparsers when setting `exit_on_error`.

This has been fixed, tests are passing and command suggestions are shown when argparse throws an error for subcommands:
<img width="1082" alt="Screen Shot 2020-12-22 at 9 30 11 PM" src="https://user-images.githubusercontent.com/7004949/102963724-b3620880-449e-11eb-86e3-4793778c0202.png">

This is definitely in better shape than where it was a couple of weeks/months ago 👍 

@efiop any issues with some of this requiring Python 3.9? How should we proceed?

Reply:
I see `sys.version_info` is used throughout the code for checking Python versions: https://github.com/iterative/dvc/search?q=%22version_info%22

Maybe this can be used to determine whether or not to initialize `DvcParser` with `exit_on_error` set but that feels wrong.

Reply:
> @efiop any issues with some of this requiring Python 3.9? How should we proceed?

@santos22 I haven't looked deep into it, so not sure what is the best solution here. We could either fix old versions or disable suggestions for older python versions, if it is too hard.